Respect non-Adwaita system theme.

This makes libadwaita more like libhandy for gtk4.

--- libadwaita-1.3.6/src/adw-style-manager.c.~1~	2023-12-01 15:39:23.000000000 +0000
+++ libadwaita-1.3.6/src/adw-style-manager.c	2024-10-26 11:57:44.261180529 +0000
@@ -239,9 +239,11 @@
   if (self->display) {
     GtkSettings *settings = gtk_settings_get_for_display (self->display);
     gboolean prefer_dark_theme;
+    char* theme_name;
 
     g_object_get (settings,
                   "gtk-application-prefer-dark-theme", &prefer_dark_theme,
+                  "gtk-theme-name", &theme_name,
                   NULL);
 
     if (prefer_dark_theme)
@@ -253,7 +255,8 @@
                              self,
                              G_CONNECT_SWAPPED);
 
-    if (!adw_is_granite_present () && !g_getenv ("GTK_THEME")) {
+    // respect non-Adwaita themes
+    if (!adw_is_granite_present () && strcasestr (g_getenv ("GTK_THEME") ?: theme_name, "default")) {
       g_object_set (gtk_settings_get_for_display (self->display),
                     "gtk-theme-name", "Adwaita-empty",
                     NULL);
@@ -269,6 +272,8 @@
                                                   GTK_STYLE_PROVIDER_PRIORITY_THEME);
     }
 
+    g_free(theme_name);
+
     self->animations_provider = gtk_css_provider_new ();
     gtk_css_provider_load_from_data (self->animations_provider,
                                      "* { transition: none; }",
